### PR TITLE
InputCommon/GCAdapter: Fix offbrand "GCAdapters".

### PR DESCRIPTION
**Pull request by jordan-woyak**
This makes my Nyko-branded GCAdapter work and probably others. Please test.

Note that `libusb_control_transfer` calls give a `LIBUSB_ERROR_BUSY` error within a hotplug callback so I've introduced an `Event` to initialize adapters in the "scanning" thread.

I'm not entirely happy with this fix but GCAdapter needs a complete rewrite. The error handling is bad. The inter-thread communication is bad. And 90% of the logic is copy-pasted to GCAdapter_Android.cpp.

This PR does not attempt to fix the Android-side of things. I can't test there.

### CODE REVIEW

**Comment by JMC47** [comment]
I *can* test GC adapter code on Android, but I don't have Nyko.

**Comment by JMC47** [comment]
This adapter here still doesn't work.

https://cdn.discordapp.com/attachments/521710974827495435/729598461900488754/unknown.png

It spams the logs with this:

17:55:603 InputCommon\GCAdapter.cpp:91 E[SI]: adapter libusb read failed: err=LIBUSB_ERROR_IO

**Comment by NikhilNarayana** [comment]
I have had two people test master and this PR with a pdobq adapter and this fix has worked for them.

Image of the adapter for reference: https://cdn.discordapp.com/attachments/363806218181017601/729876869658181702/image0.jpg

File: Source/Core/InputCommon/GCAdapter.cpp [hunk 1]
```diff
@@ -307,6 +303,10 @@ static bool CheckDeviceAccess(libusb_device* device)
       ERROR_LOG(SERIALINTERFACE, "libusb_detach_kernel_driver failed with error: %d", ret);
   }
 
+  ret = libusb_control_transfer(s_handle, 0x21, 11, 0x0001, 0, nullptr, 0, 10000);
```
Comment (iwubcode) [inline Source/Core/InputCommon/GCAdapter.cpp:306]: What are these magic numbers?  Might be worthwhile to add some constants or some comments

Comment (iwubcode) [reply Source/Core/InputCommon/GCAdapter.cpp:306]: Hmm, actually looking closer..this file, has a lot of magic numbers.

**Comment by lioncash** [comment]
@NikhilNarayana Thanks for having some testing done, even though you didn't have to!

**Comment by Skycl0ps** [comment]
This is the root for my adapter not working I believe. Idk what went down but my adapter was working fine last night and after updating today it no longer works.

**Comment by JMC47** [comment]
Could you list what adapter you are using?

**Comment by Zerogamer001** [comment]
This version of dolphin made my Mayflash Gamecube adapter stop working it keeps giving me the error "Error Opening Adapter: Pipe error". I tried the previous developmental build (version 5.0-12261) and it was working fine again, please fix this problem.

**Comment by jordan-woyak** [comment]
Issue reports can be filed here: https://bugs.dolphin-emu.org/projects/emulator/issues

**Comment by Zerogamer001** [comment]
thank you, it looks like someone already made a post about some adapters not working because of this build. hopefully it gets fixed soon

**Pull request merged**